### PR TITLE
docs: initial draft for resource labels

### PR DESCRIPTION
**Pull request by alvaromartmart**
WIP

### CODE REVIEW

**Comment by swampie** [comment]
@llewellyn-sl @mbosio85 @pditommaso @evanfloden ready to review

File: docs/resource-labels/overview.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+---
+title: Resource Labels Overview
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:2]:
```suggestion
title: Resource labels overview
```

File: docs/resource-labels/overview.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+headline: 'Resource Labels'
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:3]:
```suggestion
headline: 'Resource labels'
```

File: docs/resource-labels/overview.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+description: 'Step-by-step instructions to set-up and use Resurce Labels in Tower.'
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:4]:
```suggestion
description: 'Step-by-step instructions to set-up and use Resource labels in Tower.'
```

File: docs/resource-labels/overview.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+---
+
+## Introduction
+
+Resource labels are a declination of the general label concept, they have a different format `key=value` and a different behavior than the general labels.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:9]:
```suggestion
From version 22.3.X, Tower supports applying resource labels to compute environments and other Tower elements. This offers a flexible tagging system for annotation and tracking of the cloud services consumed by a run. 
```

File: docs/resource-labels/overview.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
+---
+title: Resource Labels Overview
+headline: 'Resource Labels'
+description: 'Step-by-step instructions to set-up and use Resurce Labels in Tower.'
+---
+
+## Introduction
+
+Resource labels are a declination of the general label concept, they have a different format `key=value` and a different behavior than the general labels. 
+Resource labels can be associated to Tower elements like a compute environment and are sent to cloud provides upon a workflow execution to tag the resources generated by it.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:10]:
```suggestion
Resource labels are sent to the service provider for each cloud compute environment in `key=value` format. They can be created, edited, and applied by a workspace admin or owner. 
```

File: docs/resource-labels/overview.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+The resource labels offer a flexible tagging system interacting with the cloud service provider. Tower resource labels  enable annotating, tracking, and categorizing cloud resources used by a workflow run.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:12]:
```suggestion
```

File: docs/resource-labels/overview.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+
+## Create and apply labels
+
+Resource labels can be created, applied and edited by a workspace admin or owner. When applying a label, users can select from existing labels or add new ones on the fly.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:17]:
```suggestion
Resource labels can be created, applied, and edited by a workspace admin or owner. When applying a label, users can select from existing labels or add new labels on-the-fly. 
```

File: docs/resource-labels/overview.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+## Create and apply labels
+
+Resource labels can be created, applied and edited by a workspace admin or owner. When applying a label, users can select from existing labels or add new ones on the fly. 
+
+![](_images/ce-resource-labels.png)
+
+### Resource labels applied to a compute environment
+Admins can assign a set of resource labels when creating a compute environment.
+All runs executed using the compute environment will be tagged with the compute enviromment resoruce labels.
+Resoruce labels applied to a compute environment are displayed in the compute environment details page.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:24]:
```suggestion
Resource labels applied to a compute environment are displayed on the compute environment details page.
```

File: docs/resource-labels/overview.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+![](_images/launch-resource-labels.png)
+
+Apply label to a compute enviromnent when adding a new compute environment to the workspace.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:28]:
```suggestion
Apply a label when adding a new compute environment to the workspace.
```

File: docs/resource-labels/overview.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+!!!warning
+    Once the compute environment is created, the resource labels cannot be edited at all.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:31]:
```suggestion
    Once the compute environment is created, its resource labels cannot be edited.
```

File: docs/resource-labels/overview.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+If a resource label as applied to a compute environment, all workflow runs of this pipeline will inherit the resource label. Likewise, all cloud resrouces generated during the workflow execution will be tagged with the same resource label.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:33]:
```suggestion
If a resource label is applied to a compute environment, all runs in that compute environment will inherit it. Likewise, all cloud resources generated during the workflow execution will be tagged with the same resource label.
```

File: docs/resource-labels/overview.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+![](_images/workflow-resource-labels.png)
+
+
+## Resource label propagation to cloud environments
+Resource labels are only available for cloud environment endowed with a resource tagging system.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:39]:
```suggestion
Resource labels are only available for cloud environments that use a resource tagging system. 
```

File: docs/resource-labels/overview.md [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+    Once the compute environment is created, the resource labels cannot be edited at all.
+
+If a resource label as applied to a compute environment, all workflow runs of this pipeline will inherit the resource label. Likewise, all cloud resrouces generated during the workflow execution will be tagged with the same resource label.
+
+![](_images/workflow-resource-labels.png)
+
+
+## Resource label propagation to cloud environments
+Resource labels are only available for cloud environment endowed with a resource tagging system. 
+Tower supports AWS, Google Life Science, Azure and Kubernetes:
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:40]:
```suggestion
Tower supports AWS, Google Life Sciences, Azure, and Kubernetes — HPC compute environments do not support resource labels. 
```

File: docs/resource-labels/overview.md [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+
+If a resource label as applied to a compute environment, all workflow runs of this pipeline will inherit the resource label. Likewise, all cloud resrouces generated during the workflow execution will be tagged with the same resource label.
+
+![](_images/workflow-resource-labels.png)
+
+
+## Resource label propagation to cloud environments
+Resource labels are only available for cloud environment endowed with a resource tagging system. 
+Tower supports AWS, Google Life Science, Azure and Kubernetes: 
+HPC compute environments cannot have resouce labels associated.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:41]:
```suggestion
```

File: docs/resource-labels/overview.md [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+
+Whenever a run is executed in a compute environment with associated resource labels, Tower takes care to propagate the labels to a set of resources, listed below, while Nextflow will distribute the labels for the resources that are spawn at runtime
+
+If the compute environment is created through Forge then compute environment creation will propagate the tags to the resources generated by the Forge execution.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:45]:
```suggestion
If the compute environment is created through Forge, the compute environment will propagate the tags to the resources generated by the Forge execution.
```

File: docs/resource-labels/overview.md [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+
+### AWS
+
+When the compute environment is created with forge then the following resources will be tagged using the labels associated with the compute environment
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:49]:
```suggestion
When the compute environment is created with Forge, the following resources will be tagged using the labels associated with the compute environment:
```

File: docs/resource-labels/overview.md [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+
+- FSX Filesystems
+- EFS Filesystems
+- Roles and Policies
+- Batch Compute Environment
+- Batch Queue(s)
+
+At execution time, when the jobs are submitted to Batch, the request are all instrumented to propagate tags to all the  instances created by the head job
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:57]:
```suggestion
At execution time, when the jobs are submitted to Batch, the requests are set up to propagate tags to all the instances created by the head job.
```

File: docs/resource-labels/overview.md [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+
+- FSX Filesystems
+- EFS Filesystems
+- Roles and Policies
+- Batch Compute Environment
+- Batch Queue(s)
+
+At execution time, when the jobs are submitted to Batch, the request are all instrumented to propagate tags to all the  instances created by the head job 
+
+### Google Life Science
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:59]:
```suggestion
### Google Life Sciences
```

File: docs/resource-labels/overview.md [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
+
+For GLS the only entity that will be tagged is the `Run Pipeline Request`: the propagation of the labels to subsequent resources during execution is delegated to Nextflow
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:61]:
```suggestion
For GLS, the only entity that will be tagged is the `Run Pipeline Request`: the propagation of the labels to subsequent resources during execution is delegated to Nextflow.
```

File: docs/resource-labels/overview.md [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+### Azure
+
+!!!warning
+The labeling system on Azure Cloud uses the naming Metadata
+
+When creating an Azure Compute Environment through Forge will add the resource labels to the Pool Parameters: this will add a set of key=value metadatas to the Azure Batch Pool.
+
+### Kubernetes
+
+Both Head Pod and Work pods specs will contains the set of labels associated with the compute environment along with the standard labels applied by Tower and Nextflow
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:72]:
```suggestion
Both the Head pod and Work pod specs will contain the set of labels associated with the compute environment in addition to the standard labels applied by Tower and Nextflow.
```

File: docs/resource-labels/overview.md [hunk 21]
```diff
@@ -0,0 +1,106 @@
...
+
+
+!!!warning
+    Currently, the tagging with resource labels is not available for the files created during a workflow execution. The cloud instances are the elements being tagged.
+
+!!!warning
+    The resource labels propagation is one-way and not synchronized with the cloud environment. This means that Tower takes care to attach tags to cloud resources but is not aware if those tags change  or are deleted directly in the cloud environment.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:79]:
```suggestion
    Resource label propagation is one-way and not synchronized with the cloud environment. This means that Tower attaches tags to cloud resources, but is not aware if those tags are changed or deleted directly in the cloud environment.
```

File: docs/resource-labels/overview.md [hunk 22]
```diff
@@ -0,0 +1,106 @@
...
+
+## Search and filter with labels
+Search and filter pipelines and workflow runs using one or more resource labels. 
+Filter and search are complementary and it works using the same keyword as for labels.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:83]:
```suggestion
Filter and search use the same keywords as labels.
```

File: docs/resource-labels/overview.md [hunk 23]
```diff
@@ -0,0 +1,106 @@
...
+The requirements are to have a `label:key=value` format. 
+
+![](_images/filter_labels.png)
+
+## Overview of resource labels in a workspace
+
+All resource labels used in a workspace can be viewed, in the workspace’s Setting screen.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:90]:
```suggestion
All resource labels used in a workspace can be viewed in the workspace’s Settings screen. 
```

File: docs/resource-labels/overview.md [hunk 24]
```diff
@@ -0,0 +1,106 @@
...
+Search and filter pipelines and workflow runs using one or more resource labels. 
+Filter and search are complementary and it works using the same keyword as for labels.
+The requirements are to have a `label:key=value` format. 
+
+![](_images/filter_labels.png)
+
+## Overview of resource labels in a workspace
+
+All resource labels used in a workspace can be viewed, in the workspace’s Setting screen. 
+Resrouce labels can be edited or deleted only by admins and only if they are not associated to **any** Tower resource.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:91]:
```suggestion
Resource labels can only be edited or deleted by admins and only if they are not already associated with **any** Tower resource.
```

File: docs/resource-labels/overview.md [hunk 25]
```diff
@@ -0,0 +1,106 @@
...
+This includes both compute environments and workflow runs.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:92]:
```suggestion
This includes both compute environments and runs.
```

File: docs/resource-labels/overview.md [hunk 26]
```diff
@@ -0,0 +1,106 @@
...
+The deletion of a resource label from a workspace has no influence whatsoever to the cloud environment.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:93]:
```suggestion
The deletion of a resource label from a workspace has no influence on the cloud environment. 
```

File: docs/resource-labels/overview.md [hunk 27]
```diff
@@ -0,0 +1,106 @@
...
+Resrouce labels can be edited or deleted only by admins and only if they are not associated to **any** Tower resource.
+This includes both compute environments and workflow runs.
+The deletion of a resource label from a workspace has no influence whatsoever to the cloud environment. 
+
+![](_images/label_management.png)
+
+## Limits
+
+!!!warning
+    Resoruce labels need to have non-null key and value. Both key and value follow the same limitations for simple labels in terms of composition and length.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:100]:
```suggestion
    Resource labels need to have non-null key and value. Both the key and value have the same limitations as simple labels for composition and length.
```

File: docs/resource-labels/overview.md [hunk 28]
```diff
@@ -0,0 +1,106 @@
...
+
+!!!warning
+    Currently, it is not possible to assign multiple resource labels with the same key but different values to the same resource, regardless if this option is supported or handled by the destination cloud provider.
+
+- A maximum of 25 resource abels can be applied to each resource.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:105]:
```suggestion
- Resource labels can have a maximum of 25 characters.
```

File: docs/resource-labels/overview.md [hunk 29]
```diff
@@ -0,0 +1,106 @@
...
+- A maximum of 100 resource labels  can be used in each workspace.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:106]:
```suggestion
- A maximum of 100 resource labels can be used in each workspace.  
```

File: docs/resource-labels/overview.md [hunk 30]
```diff
@@ -0,0 +1,104 @@
...
+title: Resource labels overview
+headline: 'Resource labels'
+description: 'Step-by-step instructions to set-up and use Resource labels in Tower.'
+---
+
+## Introduction
+
+From version 22.3, Tower supports applying resource labels to compute environments and other Tower elements. This offers a flexible tagging system for annotation and tracking of the cloud services consumed by a run. 
+Resource labels are sent to the service provider for each cloud compute environment in `key=value` format. 
+
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:11]:
```suggestion
```

File: docs/resource-labels/overview.md [hunk 31]
```diff
@@ -0,0 +1,104 @@
...
+
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:12]:
```suggestion
```

File: docs/resource-labels/overview.md [hunk 32]
```diff
@@ -0,0 +1,104 @@
...
+
+## Create and apply labels
+
+Resource labels can be created, applied, and edited by a workspace admin or owner. When applying a label, users can select from existing labels or add new labels on the fly. 
+
+![](_images/ce-resource-labels.png)
+
+### Resource labels applied to a compute environment
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:20]:
```suggestion
### Resource labels applied to a compute environment

```

File: docs/resource-labels/overview.md [hunk 33]
```diff
@@ -0,0 +1,104 @@
...
+
+Apply a label when adding a new compute environment to the workspace.
+
+!!!warning
+    Once the compute environment has been created, its resource labels cannot be edited.
+
+If a resource label is applied to a compute environment, all runs in that compute environment will inherit it. Likewise, all cloud resources generated during the workflow execution will be tagged with the same resource label.
+
+![](_images/workflow-resource-labels.png)
+
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:35]:
```suggestion
```

File: docs/resource-labels/overview.md [hunk 34]
```diff
@@ -0,0 +1,104 @@
...
+!!!warning
+The labeling system on Azure Cloud uses the naming Metadata
+
+When creating an Azure Compute Environment through Forge, resource labels are added to the Pool parameters: this will add a set of `key=value` metadata pairs to the Azure Batch Pool.
+
+### Kubernetes
+
+Both the Head pod and Work pod specs will contain the set of labels associated with the compute environment in addition to the standard labels applied by Tower and Nextflow.
+
+
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:72]:
```suggestion
```

File: docs/resource-labels/overview.md [hunk 35]
```diff
@@ -0,0 +1,104 @@
...
+!!!warning
+    Currently, tagging with resource labels is not available for the files created during a workflow execution. The cloud instances are the elements being tagged.
+
+!!!warning
+    Resource label propagation is one-way and not synchronized with the cloud environment. This means that Tower attaches tags to cloud resources, but is not aware if those tags are changed or deleted directly in the cloud environment.
+
+## Search and filter with labels
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:79]:
```suggestion
## Search and filter with labels

```

File: docs/resource-labels/overview.md [hunk 36]
```diff
@@ -0,0 +1,104 @@
...
+Search and filter pipelines and workflow runs using one or more resource labels.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:80]:
```suggestion
Search and filter pipelines and runs using one or more resource labels. 
```

File: docs/resource-labels/overview.md [hunk 37]
```diff
@@ -0,0 +1,104 @@
...
+Resource labels can only be edited or deleted by admins and only if they are not already associated with **any** Tower resource.
+This includes both compute environments and runs.
+The deletion of a resource label from a workspace has no influence on the cloud environment. 
+
+![](_images/label_management.png)
+
+## Limits
+
+!!!warning
+    Resource labels need to have non-null key and value. Both the key and value have the same limitations as simple labels for composition and length.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:98]:
```suggestion
    Resource labels need to have a non-null key and value. Both the key and value have the same limitations as simple labels for composition and length.
```

File: docs/resource-labels/overview.md [hunk 38]
```diff
@@ -0,0 +1,104 @@
...
+
+!!!warning
+    Currently, it is not possible to assign multiple resource labels, using the same key, to the same resource, regardless of whether this option is supported by the destination cloud provider.
```
Comment (llewellyn-sl) [inline docs/resource-labels/overview.md:101]:
```suggestion
    Currently, it is not possible to assign multiple resource labels, using the same key, to the same resource — regardless of whether this option is supported by the destination cloud provider.
```

**Review by llewellyn-sl — APPROVED** [review]

**Pull request merged**